presto-hudi/src/main/java/com/facebook/presto/hudi/split/HudiPartitionSplitGenerator.java

```diff
@@ -0,0 +1,156 @@
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A runnable to take partition names from a queue of partitions to process,
+ * generate Hudi splits from the partition, and add the splits to the async
+ * ConnectorSplit queue.
+ */
+public class HudiPartitionSplitGenerator
+        implements Runnable
+{
+    private static final Logger log = Logger.get(HudiPartitionSplitGenerator.class);
+
+    private final ExtendedHiveMetastore metastore;
+    private final MetastoreContext metastoreContext;
+    private final HudiTableLayoutHandle layout;
+    private final HudiTableHandle table;
+    private final Path tablePath;
+    private final HoodieTableFileSystemView fsView;
+    private final AsyncQueue<ConnectorSplit> asyncQueue;
+    private final Queue<String> concurrentPartitionQueue;
+    private final String latestInstant;
+    private final HudiSplitWeightProvider splitWeightProvider;
+
+    public HudiPartitionSplitGenerator(
+            ConnectorSession session,
+            ExtendedHiveMetastore metastore,
+            HudiTableLayoutHandle layout,
+            HoodieTableFileSystemView fsView,
+            AsyncQueue<ConnectorSplit> asyncQueue,
+            Queue<String> concurrentPartitionQueue,
+            String latestInstant)
+    {
+        this.metastore = requireNonNull(metastore, "metastore is null");
+        this.metastoreContext = toMetastoreContext(requireNonNull(session, "session is null"));
+        this.layout = requireNonNull(layout, "layout is null");
+        this.table = layout.getTable();
+        this.tablePath = new Path(table.getPath());
+        this.fsView = requireNonNull(fsView, "fsView is null");
+        this.asyncQueue = requireNonNull(asyncQueue, "asyncQueue is null");
+        this.concurrentPartitionQueue = requireNonNull(concurrentPartitionQueue, "concurrentPartitionQueue is null");
+        this.latestInstant = requireNonNull(latestInstant, "latestInstant is null");
+        this.splitWeightProvider = createSplitWeightProvider(requireNonNull(session, "session is null"));
+    }
+
+    @Override
+    public void run()
+    {
+        HoodieTimer timer = new HoodieTimer().startTimer();
+
+        while (!concurrentPartitionQueue.isEmpty()) {
+            String partitionName = concurrentPartitionQueue.poll();
+
+            if (partitionName != null) {
+                generateSplitsFromPartition(partitionName);
+            }
+        }
+        log.debug("HudiPartitionSplitGenerator %s finishes in %d ms", this, timer.endTimer());
+    }
+
+    private void generateSplitsFromPartition(String partitionName)
+    {
+        HudiPartition hudiPartition = getHudiPartition(metastore, metastoreContext, layout, partitionName);
+        Path partitionPath = new Path(hudiPartition.getStorage().getLocation());
+        String relativePartitionPath = FSUtils.getRelativePartitionPath(tablePath, partitionPath);
+        fsView.getLatestFileSlicesBeforeOrOn(relativePartitionPath, latestInstant, false)
```

Comment:
Just want to confirm efficiency in terms of RPC calls.. This does 

- (without MDT) exactly one fs.listStatus RPC call for the partition, fetching the file statuses with their lengths
- (with MDT) one read of metadata table for the partition. 

Can you confirm

Reply:
That is correct. I have also verfied through logs. Without MDT, there is one log per partition as below:
```
2023-02-16T15:21:33.679+0530	INFO	hudi-split-generator-37	org.apache.hudi.common.table.view.AbstractTableFileSystemView Building file system view for partition (inv_date_sk=2452593)
```
With MDT, exactly one log for `files` partition:
```
2023-02-16T15:15:46.722+0530	INFO	hudi-split-generator-26	org.apache.hudi.common.table.view.AbstractTableFileSystemView Building file system view for partition (files)
```